cve_bin_tool/checkers/perl.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+
+"""
+CVE checker for Perl
+References:
+https://www.cvedetails.com/vulnerability-list/vendor_id-1885/Perl.html
+"""
+from ..util import regex_find
+import re
+
+
+def guess_version(lines):
+    regex1 = [r"perl-[45]:([5]+\.[0-9]+\.[0-9]+)", r"perl-[45]:([5]+\.[0-9]+)"]
+    match = regex_find(lines, *regex1)
+    if match != "UNKNOWN":
+        return match
+    match = re.sub(
+        r"This is perl ([45])+, version ([0-9])+, subversion ([0-9])+",
+        r"\1.\2.\3",
+        lines,
```

Comment:
Just putting a review commends so you can see where the problem's happening.  re.sub() is expecting a string, not an array of strings, here.